spaceapi/src/utils.rs

```diff
@@ -3,7 +3,7 @@
 /// An ``Optional`` can contain ``Optional::Value<T>`` or ``Optional::Absent``.
 /// It is similar to an ``Option``, but ``Optional::Absent`` means it will be
 /// omitted when serialized, while ``None`` will be serialized to ``null``.
-#[derive(Debug, Copy, Clone)]
+#[derive(Clone, Copy, PartialEq, PartialOrd, Eq, Ord, Debug, Hash)]
```

Comment:
How are `Optional<T>` compared when deriving these traits? Also, why are we using both `PartialEq` and `Eq`? Do you think `Absent == Absent` should be `True`? And if we want `Eq` behavior, do we also need `PartialEq`?